src/view/invoice.js

```diff
@@ -70,4 +87,9 @@ InvoiceView.propTypes = {
   invoice: PropTypes.object.isRequired,
 };
 
+const setAmount = (fiatAmount, invoice, settings) => {
+  const satoshis = fiatToSatoshis(fiatAmount, settings);
+  const amount = toAmount(satoshis, settings.unit);
+  invoice.setAmount({ amount });
+};
```

Comment:
Since when do we put logic in react components? 🙃This logic is unit testable and needs to happen in the actions.

Reply:
Lol, true.